### PR TITLE
Avoid hasattr in `L.BatchNormalization`

### PR DESCRIPTION
**Pull request by toslunar**
See #3952.

### CODE REVIEW

**Comment by niboshi** [comment]
This is no-compat because the use of `hasattr` has been documented.

**Comment by toslunar** [comment]
The no-compat will be only between v5.0.0b2 and v5.0.0b3.  The document with `hasattr` is added in #4673 and it's not been backported to v4.

**Comment by niboshi** [comment]
I see. Then, it's no-compat.
If #4673 is to be backported, this PR must also be backported at the same time.

**Comment by kmaehashi** [comment]
I added `blocking` label. https://github.com/chainer/chainer/pull/4673#issuecomment-400558694

File: chainer/links/normalization/batch_normalization.py [hunk 1]
```diff
@@ -60,10 +60,10 @@ class BatchNormalization(link.Link):
        :func:`~chainer.functions.fixed_batch_normalization`
 
     Attributes:
-        gamma (~chainer.Variable): Scaling parameter.
-        beta (~chainer.Variable): Shifting parameter.
-        avg_mean (numpy.ndarray or cupy.ndarray): Population mean.
-        avg_var (numpy.ndarray or cupy.ndarray): Population variance.
+        ~BatchNormalization.gamma (~chainer.Variable): Scaling parameter.
+        ~BatchNormalization.beta (~chainer.Variable): Shifting parameter.
+        ~BatchNormalization.avg_mean (numpy.ndarray or cupy.ndarray): Population mean.
```
Comment (okuta) [inline chainer/links/normalization/batch_normalization.py]: Please check flake8 result.

File: chainer/links/normalization/batch_normalization.py [hunk 2]
```diff
@@ -60,10 +60,10 @@ class BatchNormalization(link.Link):
...
+        ~BatchNormalization.avg_var (numpy.ndarray or cupy.ndarray): Population variance.
```
Comment (okuta) [inline chainer/links/normalization/batch_normalization.py]: Please check flake8 result.

**Comment by okuta** [comment]
jenkins, test this please.

**Comment by toslunar** [comment]
Jenkins, test this please.

**Comment by toslunar** [comment]
Jenkins, test this please.

**Comment by okuta** [comment]
Please resolve conflict.

**Comment by toslunar** [comment]
Done.

**Comment by okuta** [comment]
LGTM!

**Comment by okuta** [comment]
Can I remove to-be-backport label?

**Comment by okuta** [comment]
OK, I wait it.
